In [0]:
import numpy as np
import pandas as pd
import scipy.stats as stats
import math
import seaborn as sns
import matplotlib.pyplot as plt
from scipy.stats import ttest_rel,ttest_ind,ttest_1samp
import statsmodels.api as sm
from   statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm
import warnings 
warnings.filterwarnings('ignore')
from scipy.stats import ttest_1samp,ttest_ind, wilcoxon,levene,bartlett,shapiro,mannwhitneyu
import numpy       as np
import pandas      as pd
import scipy.stats as stats
import warnings

Question 1. The Bangalore International Airport Ltd.(BIAL) maintains statistics for mishandled baggage per 1,000

airline passengers. In the last year 2018-19, on an average they mishandled 1.5 baggage per 1,000
customers. What is the probability that in the next 1,000 customers, BIAL will have
1. no mishandled baggage (1 MARK)
2. at least one mishandled baggage (2MARK)
3. at most 3 mishandled baggage(2MARK)

In [0]:
#Given that 
n = 1000  # Total number of passengers
mu = 1.5  # Average number of mishandled bags per 1000
p = 1.5/1000

In [83]:
# no mishandled baggage
stats.binom.cdf(0,1000,p)

0.22287902884342548

In [84]:
1-stats.binom.cdf(0,1000,p)

0.7771209711565745

In [89]:
1-stats.binom.cdf(3,1000,p)

0.06550111057355135

Question 2. The below data contains the weight of 20 random ‘LAYS’ pockets in grams.
9.0, 8.8, 10.2, 7.4,9.5, 7.0, 6.6, 9.6, 9.9, 7.6, 7.4, 11.6, 10.4, 11.1, 8.5, 10.0, 10.7, 10.3, 11.2, and 8.0
1. Is the mean weight of all(Population) LAYS pockets μ = 10grams different from the mean weight
of this sample? (3 MARKS)
2. Find the 95% confidence interval. (2 MARKS)

In [0]:
lays_data = [9.0, 8.8, 10.2, 7.4,9.5, 7.0, 6.6, 9.6, 9.9, 7.6, 7.4, 11.6, 10.4, 11.1, 8.5, 10.0, 10.7, 10.3, 11.2,8.0]

 We have to check whether mean weight of all(Population) LAYS pockets μ = 10grams different from the mean weight of this sample
For this we have to make Hypothesis Testing .
Given is the population mean mup = 10 grams.
Thus,Creating Null and Alternate Hypothesis:
1. H0: Mean weights of Population is equal to Mean weight of the sample.
2. Ha: Mean weights of Population is not equal to Mean weight of the sample.

We will be using t-test for this hypothesis as sample size is less than 30 and std is unknown.

In [5]:
stats.ttest_1samp(lays_data,10)


Ttest_1sampResult(statistic=-2.249161158076392, pvalue=0.03655562279112455)

Assuming alpha = 0.05 ,From the above we can see that pvalue is less than alpha.
1. Hence probablity of HO being true is less.Rejecting Null Hypothesis.
2. Hence Population mean is not equal to the sample mean.

In [7]:
# 95 % confidence Interval
#i.e alpha=0.05
mu = 10
sigma = np.std(lays_data)/np.sqrt(len(lays_data))
stats.norm.interval(0.95,mu,sigma)

(9.35449009435897, 10.64550990564103)

Thus for 95% confidence interval values are lying between 9.35 and 10.64 .
It depicts our range of certainity.

Question 3: . A set of final examination grades in SMDM course is normally distributed with a mean of 70 and a
standard deviation of 10. If the professor gives grade A for top 10% of the class regardless of the
scores, are you better off with a grade of 81 in this exam or 62 on a different exam, where the mean
is 62 and standard deviation is 3? Show your answer statistically and explain? [5 MARKS]

In [20]:
# We have been given that data is normally distributed with following statistical parameters:
mu_1 = 70 
sigma_1 = 10
# Also it has been given that for other test following are the statistical parameters:
mu_2 =62
sigma_2 =3

#Probablity of getting 81 in First exam:

print('Probablity of getting 81 in First exam',1-stats.norm.cdf(81,mu_1,sigma_1))
#Probablity of getting 62 on a Second Exam:
print('Probablity of getting 62 on a Second Exam:',1-stats.norm.cdf(62,mu_2,sigma_2))


Probablity of getting 81 in First exam 0.13566606094638267
Probablity of getting 62 on a Second Exam: 0.5


In [22]:
# Also lets find the confidence interval for both of the test:

#ASSUME: ALPHA=0.10 Thus 90% Confidence Interval

#Confidence Interval of getting 81 in First exam:

print('Confidence Interval of getting 81 in First exam',stats.norm.interval(0.9,mu_1,sigma_1))
#Confidence Interval of getting 62 on a Second Exam:
print('Confidence Interval of getting 62 on a Second Exam:',stats.norm.interval(0.90,mu_2,sigma_2))

Confidence Interval of getting 81 in First exam (53.55146373048527, 86.44853626951472)
Confidence Interval of getting 62 on a Second Exam: (57.06543911914558, 66.93456088085442)


### Hence we can infer that for getting A grade regardless of the mark we have to be in top 10% . Thus there is high Probality of Happening that in second test. 

Question 4: 4. A company manufacturing Cola has studied the effect of its latest advertising campaign and then
measured the number of cans of Cola purchased. A simple linear regression model was fitted for
number of cans of Cola purchased (Y) with number of advertisements (X). The fitted regression model
is Y = 3.331 + 1.711 X. The coefficient of determination was 0.619, but the adjusted R-square was
0.555.
1. Find the coefficient of correlation between X and Y (1 mark)
2. Interpret the values on regression coefficient, correlation coefficient, coefficient
 of determination and adjusted R-square (4 marks

In [0]:
# we have been given that y=3.331+1.711x 
#y= 3.331 + 1.711*x

# thus coefficient of correlation between x and y is 1.711

In [0]:
from sklearn.linear_model import  LinearRegression
linear = LinearRegression()



5. Multiple myeloma or blood plasma cancer is characterised by increased by blood vessel
formulation(angiogenesis) in the bone marrow that is a predictive factor in survival. One treatment
approach used for multiple myeloma is stem cell transplantation with the patient’s own stem cells.
the data is given below represent the bone marrow micro vessel density for patients who had a
complete response to the stem cell transplant (as measured by blood and urine tests). The
measurements were taken immediately prior to the stem cell transplant and at the time the complete
response was determined. (10 MARKS)

In [0]:
# given that :
before_treatment = [158,189,202,353,416,426,441]
after_treatment =  [284,214,101,227,290,176,290]

In [0]:
#a. At the 0.05 level of significance, is there evidence that the mean bone marrow micro vessel
#density is higher before the stem cell transplant than after the stem cell transplant? (5 MARKS)

alpha=0.05

For this check we have to do hypothesis testing :
1. H0:  bone marrow micro vessel density is not higher before the stem cell transplant than after the stem cell transplant.
2. H1:  bone marrow micro vessel density is higher before the stem cell transplant than after the stem cell transplant.

Now the data is paired as we checking the impact of treatment on the same patients.Also there are two samples.
 Thus select two paired t-test for the validation of hypothesis.

In [27]:
stats.ttest_rel(before_treatment,after_treatment)

Ttest_relResult(statistic=1.8424550291027728, pvalue=0.11498626687185708)

From the above we can see that p-value is 0.11 which is greater than our assumed alpha =0.05.
1. We fail to reject Null hypothesis.
2. Thus bone marrow micro vessel density is not higher before the stem cell transplant than after the stem cell transplant holds true.

In [31]:
#Interpret the meaning of test statistic and p-value from the result.
t_stats = stats.ttest_rel(before_treatment,after_treatment)[0]
t_critical =  stats.t.ppf(0.95,12)
print('t_critical',t_critical)

t_critical 1.782287555649159


### Thus we can infer that from above that test_statistics is greater than test_critical after selection of t-test. This can be interpreted in our decision making that we are failing to reject null Hypothesis. 
### Similary,p-value is coming out to be greater than alpha taken which leads us to interpret that probablity of H0 being true holds true and there is no increase in bone marrow density even after the treatement.

In [34]:
#c. Construct and interpret the 95% confidence interval estimation of the mean difference in bone
#marrow micro vessel density before and after stem cell transplant. (2 MARKS)


#Before Treatment 
mu_before = np.mean(before_treatment)
standard_error_before = np.std(before_treatment)/np.sqrt(len(before_treatment)-1)
stats.norm.interval(0.95,mu_before,standard_error_before)

(219.87531179993334, 404.410402485781)

In [35]:
#After Treatment 
mu_after = np.mean(after_treatment)
standard_error_after = np.std(after_treatment)/np.sqrt(len(after_treatment)-1)
stats.norm.interval(0.95,mu_after,standard_error_after)

(173.76988854574063, 278.2301114542594)

After,constructing and observing 95% confidence intervals for patients measurements before and after treatment we can interpret that:
1. Certainity lies between measurements 219.87 to 404.41 before treatment
2. Certainity lies changes after treatement to the measurements 173.76 to 278.23

Question 6. The mass of N1=20 acorns from oak trees up wind from a coal power plant and N2=30 acorns from
oak trees downwind from the same coal power plant are measured. Are the acorns from trees
downwind less massive then the ones from up wind? The sample sizes are not equal but we will
assume that the population variance of sample 1 and sample 2 are equal. alpha = 0.05,
Samples are given below. State the Null and Alternate hypothesis. Which test would be used here?
Do you reject or retain the H0 based on your test? Find and interpret the t-statistic, confidence
Interval and p-value. (10 MARKS)
1. Sample up wind:x1 = [10.8, 10.0, 8.2, 9.9, 11.6, 10.1, 11.3, 10.3, 10.7, 9.7, 7.8, 9.6, 9.7, 11.6,10.3, 9.8, 12.3, 11.0, 10.4, 10.4]
2. sample downwind: x2 = [7.8, 7.5, 9.5, 11.7, 8.1, 8.8, 8.8, 7.7, 9.7, 7.0, 9.0, 9.7, 11.3, 8.7, 8.8,10.9, 10.3, 9.6, 8.4, 6.6, 7.2, 7.6, 11.5, 6.6, 8.6, 10.5, 8.4, 8.5, 10.2, 9.2]


In [0]:
# Given that:
 x1 = [10.8, 10.0, 8.2, 9.9, 11.6, 10.1, 11.3, 10.3, 10.7, 9.7, 7.8, 9.6, 9.7, 11.6,10.3, 9.8, 12.3, 11.0, 10.4, 10.4]
x2  = [7.8, 7.5, 9.5, 11.7, 8.1, 8.8, 8.8, 7.7, 9.7, 7.0, 9.0, 9.7, 11.3, 8.7, 8.8,10.9, 10.3, 9.6, 8.4, 6.6, 7.2, 7.6, 11.5, 6.6, 8.6, 10.5, 8.4, 8.5, 10.2, 9.2]

In [42]:
# Test of Normality 
# Assume alpha =0.05
shapiro(x1),shapiro(x2)

((0.9515844583511353, 0.3917829096317291),
 (0.9714616537094116, 0.579898476600647))

### we can see that Data is Normal as p value is greater than alpha and H0 hold true . Unable to Reject Null Hypothesis for Normality.As given that variance is assumed to be equal thus there is no need for Test of Variance.

In [39]:
# Test of Normality
shapiro(x2)

(0.9714616537094116, 0.579898476600647)

In [0]:
#State the Null and Alternate hypothesis


### H0: acorns from trees downwind are not less massive then the ones from up wind
### Ha: acorns from trees downwind are less massive then the ones from up wind

###We have given number of samples as 30 and also population standard error is unknown. There are two independent samples .
###Thus we will go with two samples unpaired t test.

In [44]:
print(stats.ttest_ind(x1,x2))
# size  = len(x1)+len(x2)-2
print('t_critical',stats.t.ppf(0.95,48))

Ttest_indResult(statistic=3.5981947686898033, pvalue=0.0007560337478801464)
t_critical 1.6772241953450393


In [47]:
mu_x1 = np.mean(x1)
standard_error_x1 = np.std(x1)/np.sqrt(len(x1)-1)
stats.norm.interval(0.95,mu_x1,standard_error_x1)

(9.806660119026201, 10.743339880973803)

In [46]:
mu_x2 = np.mean(x2)
standard_error_x2 = np.std(x2)/np.sqrt(len(x2)-1)
stats.norm.interval(0.95,mu_x2,standard_error_x2)

(8.43572255355793, 9.444277446442069)

### From above we can see that:
1. t_stat> t_critical  # Reject Null Hypothesis
2. pvalue< 0.05      # Reject Null Hypothesis

Thus Probablity of H0 is very less likely and acorns from trees downwind are less massive then the ones from up wind.
After,constructing and observing 95% confidence intervals for patients measurements before and after treatment we can interpret that:

Certainity lies between measurements 9.86 to 10.74 before treatment
Certainity lies changes after treatement to the measurements 8.43 to 9.44


7. a) The below data is scores of ‘BRICS’ countries on ‘Starting a business’. It is collected from
‘Doingbusineess.org’, a part of World Bank Group. Is the score of ‘Starting a business depends on the
country or time or both?
1. State your hypothesis statements and level of significance (2 MARKS)
2. State which test do you apply assuming the ‘score’ follows normal distribution and determine the
test results. (4 MARKS)
3. Interpret the results obtained (2 MARKS)


In [0]:
# Hypothesis
 # Ho :  Starting a business does not depends on the country or time or both
 # Ha :  Starting a business depends on the country or time or both


# Assuming alpha =0.05

In [0]:
# For the stated Hypothesis as there are multiple category and we can apply Two way Anova can be applied

In [0]:
df = {'Country':['Brazil','Russia','India','China','South Africa'],
                 '2018':[81.3,93.1,73.9,94.1,81.2],
      '2019':[80.3,93,81,93.4,81.2],
      '2020':[64.8,93,81.6,85.4,80]}

rdf = pd.DataFrame(df)
rdf
x1 = rdf['2018']
x2 = rdf['2019']

x3 = rdf['2020']
cou = rdf['Country']
y=x1+x2+x3

In [92]:
import statsmodels.formula.api as smf
model = smf.ols('y~x1+x2+x3',rdf).fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       1.000
Model:                            OLS   Adj. R-squared:                  1.000
Method:                 Least Squares   F-statistic:                 1.210e+27
Date:                Thu, 23 Jan 2020   Prob (F-statistic):           2.11e-14
Time:                        07:32:36   Log-Likelihood:                 136.38
No. Observations:                   5   AIC:                            -264.8
Df Residuals:                       1   BIC:                            -266.3
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept  -2.274e-13   6.14e-12     -0.037      0.976   -7.82e-11    7.78e-11
x1             1.0000   1.61e-13   6.22e+12      0.000       1.000       1.000
x2             1.0000   2.61e-13   3.84e+12      0.000       1.000       1.000
x3             1.0000   7.43e-14   1.35e+13      0.000       1.000       1.000
==============================================================================
Omnibus:                          nan   Durbin-Watson:                   0.071
Prob(Omnibus):                    nan   Jarque-Bera (JB):                0.638
Skew:                           0.000   Prob(JB):                        0.727
Kurtosis:                       1.250   Cond. No.                     2.60e+03
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 2.6e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Thus there is correlation for year.

b) Four brands of bulbs were tested for their life time (in ‘000 hours). From the following data
obtained, test whether the life duration of the bulbs is significantly different. If it is significant,
which pair of methods (Make type) has significantly different (Post hoc analysis) And write your
inferences. (7 marks)

In [0]:
# Given That
make_1 = [20,23,18,17]
make_2 = [19,15,17,20,16]
make_3 = [21,19,20,17,16]
make_4 = [15,17,16,18]

In [0]:
# H0: Life duration of bulbs are not significantly different
# Ha : Life duration of bulbs are  significantly different

# Assuming significance level to be 0.05

In [59]:
shapiro(make_1),shapiro(make_2),shapiro(make_3),shapiro(make_4)
# All of them are normal.

((0.9456305503845215, 0.6889368295669556),
 (0.9523516893386841, 0.753974437713623),
 (0.9523513913154602, 0.7539724707603455),
 (0.9929120540618896, 0.97187739610672))

In [61]:
levene(make_1,make_2,make_3,make_4)
# Variance are differnt

LeveneResult(statistic=0.5483184333759047, pvalue=0.6574155870897542)

In [62]:
# Selecting f test for the samples as there are four samples with diffeerent lenghts


stats.f_oneway(make_1,make_2,make_3,make_4)

F_onewayResult(statistic=1.6688741721854305, pvalue=0.21909520543565303)

# From the above we can see that p value is greater than 0.05 assumed alpha . Thus We are unable to reject Null hypothesis.

In [66]:

# No of columns-1 =3
# no of observation-number of columns = ddof
stats.f.isf(0.05, 3,14)  # F-critical

3.3438886781189123

Thus we are unable to reject null hypothesis.Life Duration of Bulbs are not significantly different.

d)
d. The following data relates to the scores obtained by a salesman of a company in an intelligence
test and their weekly sales (in Rs.1000’s):
Salesman: A B C D E F G H I
Test Score: 50 60 50 60 80 50 80 40 70
Weekly Sales: 30 60 40 50 60 30 70 50 60
1. Obtain the regression equation of sales on intelligence test scores of the salesman. and
write your inferences on the results of regression model. (10 marks)
2. If the intelligence test score of a salesman is 65, what would be his expected weekly sales?

In [69]:
df = {'Salesman':['A','B','C','D','E','F','G','H','I'],
                 'Test Score':[50,60,50,60,80,50,80,40,70],
      'Weekly Sales': [30,60,40,50,60,30,70,50,60]}

rdf = pd.DataFrame(df)
rdf

Salesman  Test Score  Weekly Sales
0        A          50            30
1        B          60            60
2        C          50            40
3        D          60            50
4        E          80            60
5        F          50            30
6        G          80            70
7        H          40            50
8        I          70            60

In [77]:
import statsmodels.formula.api as smf
model = smf.ols('y~x',df).fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.563
Model:                            OLS   Adj. R-squared:                  0.500
Method:                 Least Squares   F-statistic:                     9.000
Date:                Thu, 23 Jan 2020   Prob (F-statistic):             0.0199
Time:                        07:24:47   Log-Likelihood:                -32.363
No. Observations:                   9   AIC:                             68.73
Df Residuals:                       7   BIC:                             69.12
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      5.0000     15.366      0.325      0.754     -31.335      41.335
x              0.7500      0.250      3.000      0.020       0.159       1.341
==============================================================================
Omnibus:                        0.173   Durbin-Watson:                   1.875
Prob(Omnibus):                  0.917   Jarque-Bera (JB):                0.358
Skew:                           0.076   Prob(JB):                        0.836
Kurtosis:                       2.035   Cond. No.                         283.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

# We have got the coefficient of determination as 0.53
# It has good correlation with x.

In [72]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
y= rdf['Weekly Sales']
x= rdf[['Test Score']]

x_train,x_test,y_train,y_test = train_test_split(x,y,test_size = 0.3)
lr = LinearRegression()
lr.fit(x_train,y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [73]:
lr.intercept_

21.666666666666664

In [75]:
lr.coef_[0]

0.5

In [0]:
# y = 21.66+0.5*x
# Weekly Sales = 21.66+ 0.5*Test SCore

In [76]:
lr.intercept_+0.5*65  # Value of Weekly Sales.

54.166666666666664